src/entity.ts

```diff
@@ -607,6 +591,38 @@ export namespace entity {
       return valueProto;
     }
 
+    if (typeof value === 'number') {
+      const integerOutOfBoundsWarning =
+        "'IntegerOutOfBoundsWarning: the value for '" +
+        property +
+        "' property is outside of bounds of a JavaScript Number.\n" +
+        "Use 'Datastore.int(<integer_value_as_string>)' to preserve accuracy during the upload.";
+
+      const typeCastWarning =
+        "TypeCastWarning: the value for '" +
+        property +
+        "' property is a JavaScript Number.\n" +
+        "Use 'Datastore.int(<integer_value_as_string>)' or " +
+        "'Datastore.double(<double_value_as_string>)' to preserve consistent " +
+        'Datastore types during the upload.';
+
+      if (Number.isInteger(value)) {
+        if (!Number.isSafeInteger(value)) {
+          process.emitWarning(integerOutOfBoundsWarning);
+        } else {
+          process.emitWarning(typeCastWarning);
+        }
+        value = new entity.Int(value);
+        valueProto.integerValue = value.value;
+        return valueProto;
+      } else {
+        process.emitWarning(typeCastWarning);
```

Comment:
As explained in my previous comment if would be nice to factor that up ?

Reply:
Resolving as the splitting is only due to the other discussion, would be refactored if we went to raising two warning simultaneously.